lib/spack/spack/test/install.py

```diff
@@ -161,6 +161,19 @@ def test_install_dependency_symlinks_pkg(
     assert os.path.isdir(dependency_dir)
 
 
+def test_install_times(
+        install_mockery, mock_fetch, mutable_mock_repo):
+    """Test install times added."""
+    spec = Spec('flatten-deps')
+    spec.concretize()
+    pkg = spec.package
+    pkg.do_install()
+
+    # Ensure dependency directory exists after the installation.
+    install_times = os.path.join(pkg.prefix, ".spack", 'install_times.json')
+    assert os.path.isfile(install_times)
```

Comment:
last thing -- can you add a test with a package that has multiple phases, and check the contents of the file?  i.e.:

- all phases are present
- total is present
- total == sum(phase times) (I guess use some epsilon since you're using floats)

Reply:
How do I simulate a package with multiple stages in the test? Even when I do cmake it only runs an install.

Reply:
I'll try doing an install from the cmd install.py test function.

Reply:
hmm, yeah can't figure it out.

Reply:
Look at `var/spack/repos/builtin.mock/packages/dev-build-test-install/package.py` -- make your own package in `builtin.mock` with custom phases if you want to.

Tests need to use the `mock_packages` fixture to make the default repo `builtin.mock` -- but the `install_mockery` fixture covers that for you already.

Reply:
Ahhh that's where they are hiding! :open_mouth: